dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/common/DefaultRequestInfoService.java

```diff
@@ -51,7 +57,36 @@
      */
     public void setCurrentInfo( RequestInfo info )
     {
+        info = sanitised( info );
         currentInfo.set( info );
+        if ( info == null )
+        {
+            MDC.remove( X_REQUEST_ID );
+            return;
+        }
+        String xRequestID = info.getHeaderXRequestID();
+        if ( xRequestID == null )
+        {
+            MDC.remove( X_REQUEST_ID );
+        }
+        else
+        {
+            MDC.put( X_REQUEST_ID, xRequestID );
+        }
+    }
+
+    private RequestInfo sanitised( RequestInfo info )
+    {
+        if ( info == null )
+        {
+            return null;
+        }
+        String xRequestID = info.getHeaderXRequestID();
+        if ( xRequestID != null && !xRequestID.matches( "[-_a-zA-Z0-9]{1,36}" ) )
```

Comment:
Centralize the regex to a method and javadoc and a test.

Reply:
done